be/src/exec/avro_scanner.cpp

```diff
@@ -362,8 +362,15 @@ Status AvroScanner::_construct_row_without_jsonpath(const avro_value_t& avro_val
         }
 
         auto& column = chunk->get_column_by_slot_id(slot_info.id_);
+        // We should expand the union type.
+        avro_value_t& cur_value = element_value;
+        if (UNLIKELY(avro_value_get_type(&cur_value) == AVRO_UNION)) {
+            avro_value_t branch;
+            RETURN_IF_ERROR(_handle_union(cur_value, branch));
```

Comment:
We will pass a pointer to a function to let it be changed. If you pass a reference, it's better to define it constant like `const avro_value_t&`

Reply:
> We will pass a pointer to a function to let it be changed. If you pass a reference, it's better to define it constant like `const avro_value_t&`

updated